### PR TITLE
Add a dump option to dump a fusion IR graph

### PR DESCRIPTION
**Pull request by naoyam**
`NVFUSER_DUMP=fusion_ir_graph` saves the dot representation of a fusion before lowering to a file named like `__tmp_fusion_ir_graph_inner_persistent_f0_c1_r0_g0.dot`.

Example visualization: https://github.com/NVIDIA/Fuser/issues/3498#issuecomment-2549421686

### CODE REVIEW

File: csrc/options.cpp [hunk 1]
```diff
@@ -112,12 +112,13 @@ std::unordered_map<DebugDumpOption, std::vector<std::string>> Options<
...
-      {"fusion_ir_concretized", DebugDumpOption::FusionIrConcretized},
-      {"fusion_ir_preseg", DebugDumpOption::FusionIrPreseg},
-      {"fusion_ir_presched", DebugDumpOption::FusionIrPresched},
       {"fusion_ir", DebugDumpOption::FusionIr},
+      {"fusion_ir_concretized", DebugDumpOption::FusionIrConcretized},
+      {"fusion_ir_graph", DebugDumpOption::FusionIrGraph},
       {"fusion_ir_math", DebugDumpOption::FusionIrMath},
+      {"fusion_ir_original", DebugDumpOption::FusionIrOriginal},
+      {"fusion_ir_presched", DebugDumpOption::FusionIrPresched},
+      {"fusion_ir_preseg", DebugDumpOption::FusionIrPreseg},
```
Comment (naoyam) [inline csrc/options.cpp:121]: Just changed to the alphabetical ordering

**Comment by naoyam** [comment]
!build

File: csrc/runtime/executor.cpp [hunk 2]
```diff
@@ -304,6 +305,9 @@ void KernelExecutor::compile(
   NVF_ERROR(
       !fusion->outputs().empty(), "No output found for this kernel, aborting.");
 
+  createKernelId(
```
Comment (wujingyue) [inline csrc/runtime/executor.cpp:308]: Remove https://github.com/NVIDIA/Fuser/blob/d62322174e7439ab4623407b6eab3e13667b0921/csrc/runtime/executor.cpp#L421-L422? I don't think there's value to create twice.

**Review by wujingyue — APPROVED** [review]

File: csrc/runtime/executor.cpp (hunk 2 above)
Comment (naoyam) [reply csrc/runtime/executor.cpp:308]: I just moved it here.

I also am not sure if this should be a stateful operation. It seems just returning a string seems sufficient, but didn't bother to do a further cleanup.

**Pull request merged**